src/test/java/pl/jakubkozlowski/learning/firststeps/controller/ChampionControllerTest.java

```diff
@@ -0,0 +1,123 @@
+package pl.jakubkozlowski.learning.firststeps.controller;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import pl.jakubkozlowski.learning.firststeps.model.ChampionDTO;
+import pl.jakubkozlowski.learning.firststeps.service.ChampionService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static pl.jakubkozlowski.learning.firststeps.descriptor.TestDescriptor.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ChampionController.class)
+public class ChampionControllerTest {
+
+
+    @MockBean
+    public ChampionService championService;
+
+    @Autowired
+    private MockMvc mvc;
+
+    private ObjectMapper objectMapper;
+
+    private ChampionDTO championDTOAatrox;
+    private ChampionDTO championDTOAhri;
+    private ChampionDTO championDTOAnivia;
+
+    private List<ChampionDTO> championDTOList;
+
+    @Before
+    public void setUp() throws Exception {
+
+        championDTOAatrox = new ChampionDTO(ID_1, AATROX);
+        championDTOAhri = new ChampionDTO(ID_2, AHRI);
+        championDTOAnivia = new ChampionDTO(ID_3, ANIVIA);
+
+        championDTOList = new ArrayList<>(3);
+        championDTOList = Arrays.asList(championDTOAatrox, championDTOAhri, championDTOAnivia);
+
+        objectMapper = new ObjectMapper();
+
+        Mockito.when(championService.findAll()).thenReturn(championDTOList);
+        Mockito.when(championService.findById(ID_1)).thenReturn(championDTOAatrox);
+
+    }
+
+    @Test
+    public void whenFindAll_thenReturnJsonArrayOfChampions() throws Exception {
+
+
+        String content = mvc.perform(get("/champions")
```

Comment:
use descriptor constant for path. This way if you change champions path you will still have valid test